client/js/store/statement/SplitStatementStore.js

```diff
@@ -50,7 +50,7 @@ const SplitStatementStore = {
     deleteSegment (state, id) {
       const index = state.segments.findIndex((el) => el.id === id)
       if (index >= 0) {
-        delete state.segments[index]
+        state.segments.splice(index, 1)
```

Comment:
This reminds me on a similar problem i was facing just a few days ago in DiplanWissen. What type is `state.segments`, Object or Array? If it is an Object, operating on its keys as if in were the numeric keys of an array would be incorrect.

After being hinted on that topic, i eventually [landed on this solution.](https://www.dev.diplanung.de/DefaultCollection/DiPlanWissen/_git/diplan-wissen/pullrequest/8447)